File: mobile/lib/shared/models/service.dart
class Service {
  final String id;
  final String? masterId;
  final String name;
  final String? description;
  final double price;
  final int durationMinutes;
  final bool isActive;

  Service({
    required this.id,
    this.masterId,
    required this.name,
    this.description,
    required this.price,
    required this.durationMinutes,
    this.isActive = true,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['id'] as String,
      masterId: json['master_id'] as String?,
      name: json['name'] as String,
      description: json['description'] as String?,
      price: (json['price'] as num).toDouble(),
      durationMinutes: json['duration_minutes'] as int,
      isActive: json['is_active'] as bool? ?? true,
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    'description': description,
    'price': price,
    'duration_minutes': durationMinutes,
  };
}

File: backend/app/models/service.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import TYPE_CHECKING, List, Optional

from sqlalchemy import (
    Boolean,
    DateTime,
    ForeignKey,
    Index,
    Integer,
    Numeric,
    String,
    Text,
    func,
)
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import BaseModel

if TYPE_CHECKING:
    from app.models.booking import Booking
    from app.models.master import MasterProfile


class Service(BaseModel):
    __tablename__ = "services"

    master_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("master_profiles.id", ondelete="CASCADE"),
        nullable=False,
    )
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    description: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    price: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    duration_minutes: Mapped[int] = mapped_column(Integer, nullable=False)
    is_active: Mapped[bool] = mapped_column(
        Boolean, server_default="true", nullable=False
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )

    # Relationships
    master: Mapped["MasterProfile"] = relationship(
        "MasterProfile", back_populates="services"
    )
    bookings: Mapped[List["Booking"]] = relationship(
        "Booking", back_populates="service"
    )

    __table_args__ = (
        Index("ix_services_master_id", "master_id"),
        Index("ix_services_price", "price"),
    )

    def __repr__(self) -> str:
        return f"<Service {self.id} {self.name}>"
